rust/src/lib/nispor/linux_bridge.rs

```diff
@@ -1,50 +1,153 @@
 }
 
 fn np_bridge_options_to_nmstate(
     np_iface: &nispor::Iface,
 ) -> LinuxBridgeOptions {
     let mut options = LinuxBridgeOptions::default();
     if let Some(ref np_bridge) = &np_iface.bridge {
-        options.stp = Some(LinuxBridgeStpOptions {
-            enabled: Some(
-                [
-                    Some(nispor::BridgeStpState::KernelStp),
-                    Some(nispor::BridgeStpState::UserStp),
-                ]
-                .contains(&np_bridge.stp_state),
-            ),
-        });
+        options.stp = Some(get_stp_options(np_bridge));
+        options.gc_timer = np_bridge.gc_timer;
+        options.group_addr = np_bridge
+            .group_addr
+            .as_ref()
+            .map(|addr| addr.to_uppercase());
+        options.group_forward_mask = np_bridge.group_fwd_mask;
+        options.group_fwd_mask = np_bridge.group_fwd_mask;
+        options.hash_max = np_bridge.multicast_hash_max;
+        options.hello_timer = np_bridge.hello_timer;
+        options.mac_ageing_time = np_bridge.ageing_time.map(devide_by_user_hz);
+        options.multicast_last_member_count =
+            np_bridge.multicast_last_member_count;
+        options.multicast_last_member_interval =
+            np_bridge.multicast_last_member_interval;
+        options.multicast_membership_interval =
+            np_bridge.multicast_membership_interval;
+        options.multicast_querier = np_bridge.multicast_querier;
+        options.multicast_querier_interval =
+            np_bridge.multicast_querier_interval;
+        options.multicast_query_interval = np_bridge.multicast_query_interval;
+        options.multicast_query_response_interval =
+            np_bridge.multicast_query_response_interval;
+        options.multicast_query_use_ifaddr =
+            np_bridge.multicast_query_use_ifaddr;
+        options.multicast_router =
+            np_bridge.multicast_router.as_ref().and_then(|r| match r {
+                nispor::BridgePortMulticastRouterType::Disabled => {
+                    Some(LinuxBridgeMulticastRouterType::Disabled)
+                }
+                nispor::BridgePortMulticastRouterType::TempQuery => {
+                    Some(LinuxBridgeMulticastRouterType::Auto)
+                }
+                nispor::BridgePortMulticastRouterType::Perm => {
+                    Some(LinuxBridgeMulticastRouterType::Enabled)
+                }
+                _ => {
+                    warn!("Unsupported linux bridge multicast router {:?}", r);
+                    None
+                }
+            });
+        options.multicast_snooping = np_bridge.multicast_snooping;
+        options.multicast_startup_query_count =
+            np_bridge.multicast_startup_query_count;
+        options.multicast_startup_query_interval =
+            np_bridge.multicast_startup_query_interval;
     }
     options
 }
+
+fn devide_by_user_hz(v: u32) -> u32 {
+    let user_hz = unsafe { libc::sysconf(libc::_SC_CLK_TCK) } as u32;
```

Comment:
Please leave a safety note for future reference.

Reply:
Commented.